### PR TITLE
32 features section

### PR DESCRIPTION
**Pull request by mohitgusain**
#32 

1. Features page UI and dynamic section

### CODE REVIEW

File: public/wp-content/themes/glific/functions.php [hunk 1]
```diff
@@ -37,3 +37,28 @@ function glific_theme_setup(){
...
+	$shortUrlRegex = '/youtu.be\/([a-zA-Z0-9_]+)\??/i';
+	$longUrlRegex = '/youtube.com\/((?:embed)|(?:watch))((?:\?v\=)|(?:\/))(\w+)/i';
+	if (preg_match($longUrlRegex, $url, $matches)) {
+		$youtube_id = $matches[count($matches) - 1];
+	}
+	if (preg_match($shortUrlRegex, $url, $matches)) {
+		$youtube_id = $matches[count($matches) - 1];
+	}
+
+	return 'https://img.youtube.com/vi/'.$youtube_id.'/0.jpg';
```
Comment (Satendra-SR) [inline public/wp-content/themes/glific/functions.php:51]: What if both preg_match() doesn't match anything?

File: public/wp-content/themes/glific/page-features.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+						if($key == 0) :?>
+							<h3 class="fz-24 leading-35 text-theme-primary font-heebo-regular position-absolute top-n12 top-xl-n17 ml-md-8">Latest demo video</h3>
+						<?php endif; ?>
+						<iframe class="embed-responsive-item w-full rounded-30 h-full border-0" src="<?php echo $embeded_video_url; ?>" allowfullscreen></iframe>
+						<p class="font-heebo-regular fz-16 leading-24 mb-0 text-right mr-xl-10 mt-6 mt-xl-7 position-absolute right-0 bottom-n9 bottom-xl-n12"><?php echo $video['time']; ?></p>
+					</div>
+
+				</div>
+				<?php endforeach; ?>
+				<div class="d-flex flex-row flex-md-column h-md-264 w-full w-md-182 w-xl-315 h-xl-456 overflow-auto mx-auto mx-md-0 mt-15 box-shadow-dark-inset-10 py-10 box-shadow-dark-inset-10 align-items-md-center pt-md-6 mt-md-0 ml-md-auto">
```
Comment (Satendra-SR) [inline public/wp-content/themes/glific/page-features.php]:
```suggestion
				<div class="d-flex flex-row flex-md-column h-md-264 w-full w-md-182 w-xl-315 h-xl-456 overflow-auto mx-auto mx-md-0 mt-15 box-shadow-dark-inset-10 py-10  align-items-md-center pt-md-6 mt-md-0 ml-md-auto">
```

File: public/wp-content/themes/glific/page-features.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	<?php $features_section = get_field('features');
+	if (!empty($features_section)) : ?>
+	<div class="features-section pt-18 py-md-26">
+		<h3 class="text-theme-primary fz-28 leading-33 fz-xl-36 leading-xl-43 font-heebo-bold text-center mb-26"><?php echo $features_section['heading']; ?></h3>
+		<div class="w-328 w-md-752 w-xl-1054 mx-auto">
+			<?php foreach( $features_section['sections'] as $key => $section ) :
+				if ( $key%2 == 0 ) {
+					$border_class = 'rounded-bottom-left-30';
+					$class ="flex-xl-row";
+				}else{
```
Comment (Satendra-SR) [inline public/wp-content/themes/glific/page-features.php]: Format?

File: public/wp-content/themes/glific/page-features.php [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+					<h5 class="font-heebo-medium fz-24 leading-35 text-white mb-0"><?php echo $section['content']['heading'] ;?></h5>
+					<p class="font-heebo-regular fz-18 leading-27 text-white mb-0 mt-6"><?php echo $section['content']['content'] ;?></p>
+				</div>
+			</div>
+			<?php endforeach; ?>
+		</div>
+	</div>
+	<?php endif; ?>
+</div>
+	<?php
```
Comment (Satendra-SR) [inline public/wp-content/themes/glific/page-features.php]: Please use [PHP FMT](https://packagecontrol.io/packages/phpfmt).

File: public/wp-content/themes/glific/page-features.php [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+				<div class="mt-6 mt-xl-0 rounded-top-30 bg-theme-secondary border-5 border-theme-primary border p-6 w-md-512 ml-md-auto ml-xl-0 h-md-242 <?php echo $border_class ; ?>">
+					<h5 class="font-heebo-medium fz-24 leading-35 text-white mb-0"><?php echo $section['content']['heading'] ;?></h5>
+					<p class="font-heebo-regular fz-18 leading-27 text-white mb-0 mt-6"><?php echo $section['content']['content'] ;?></p>
+				</div>
+			</div>
+			<?php endforeach; ?>
+		</div>
+	</div>
+	<?php endif; ?>
+</div>
```
Comment (Satendra-SR) [inline public/wp-content/themes/glific/page-features.php:77]: Seems some formatting issue, please check if it is just one tab space.

**Review by Satendra-SR — APPROVED** [review]
@mohitgusain Approving with few minor changes/questions. Have a look

File: public/wp-content/themes/glific/functions.php (hunk 1 above)
Comment (mohitgusain) [reply public/wp-content/themes/glific/functions.php:51]: It is checking either embeded url or the youtube ure, will returent the result accordingly

**Pull request merged**